src/corppa/poetry_detection/poem_excerpts.py

```diff
@@ -0,0 +1,56 @@
+"""
+Custom data type for poetry excerpts identified with the text of PPA pages.
+"""
+
+from dataclasses import asdict, dataclass
+from typing import Any, Optional
+
+
+@dataclass(kw_only=True)
+class PoemExcerpt:
+    """
+    A poem excerpt data type representing an identified poem excerpt within a
+    PPA page
+    """
+
+    # PPA page related
+    page_id: str
+    ppa_span_start: int
+    ppa_span_end: int
+    ppa_span_text: str
+    # Reference poem related
+    poem_id: str
+    ref_corpus: str
+    ref_span_start: Optional[int] = None
+    ref_span_end: Optional[int] = None
+    ref_span_text: Optional[str] = None
+    # Methods & additional notes
+    detection_methods: set[str]
+    identification_methods: set[str]
+    notes: Optional[str] = None
+
+    def __post_init__(self):
+        # Check PPA span indices
+        if self.ppa_span_end <= self.ppa_span_start:
+            raise ValueError("PPA span's start index must be less than end index")
+        # Check that both reference span indicies are set or unset
+        if (self.ref_span_start is None) ^ (self.ref_span_end is None):
+            raise ValueError("Reference span's start and end index must both be set")
+        # Check reference span indices if set
+        if self.ref_span_end is not None and self.ref_span_end <= self.ref_span_start:
+            raise ValueError("Reference span's start index must be less than end index")
```

Comment:
Could be helpful to include the values in the error message. (I guess this change will require revising the test message)

```suggestion
            raise ValueError(f"Reference span's start index {self.ref_span_start} must be less than end index {self.ref.span_end}")
```

Reply:
I'll make a commit that includes this suggestion and updates the test accordingly. I'll do the same for the page span indices as well.